### ISSUE TITLE
bigip_ltm_policy fails with "does not have update access to partition (Common)"

### ISSUE THREAD

**Issue by benjfranklin** [F5Networks/terraform-provider-bigip#333]
I am using Terraform to deploy a policy to a partition as follows:

```
resource "bigip_ltm_policy" "hostheaders" {
  name           = "/Axiom/Axiom_HostHeader_Routing_Policy"
  strategy       = "first-match"
  requires       = ["http"]
  controls       = ["forwarding"]

  rule {
      name = "Axiom_Environment_APP1_Redirect"

      condition {
        http_host = true
        contains = true
        value = true
        values = ["acme.example.com"]
        request = true
      }

      action {
        forward = true
        pool    = "/Axiom/Axiom_Environment_APP1_Pool"
        request = true
      }
    }
  }

  depends_on = [bigip_ltm_pool.app1, bigip_ltm_pool.rhel1]
```

The user account I am using has "Manager" access to the /Axiom partition

The deployment fails as follows:

```
Error: 01070824:3: Write Access Denied: user (************) does not have update access to partition (Common), check your current "update/write" partition settings

  on ../CloudModules/axiom-environments/f5_bigip.tf line 129, in resource "bigip_ltm_policy" "hostheaders":
 129: resource "bigip_ltm_policy" "hostheaders" {
```

Note1: I have also tried using a rule-name that is partition specific e.g. 
```
  rule {
      name = "/Axiom/Axiom_Environment_APP1_Redirect"
```
This also fails.

Note2:
All other F5 provider resources (e.g. irules, pools, virtuals) deploy successfully with the same user account in the same partition.

**Comment by nmenant** [comment]
may you tell us: 

- which BIG-IP provider version you are using ? 
- version of your BIG-IP? 

I'm hitting another issue than yours for now. 

To try to save some time, could you try to do the following ? 
- Re-run this terraform with admin privilegies (if you have them - it means full read/write access on ALL partitions including common) 
- See if it is successful or not. 

when i switch to admin, i'm getting the following error message: 
``` 
bigip_ltm_policy.test-policy: Refreshing state... [id=/Axiom/my_policy]
bigip_ltm_policy.test-policy: Creating...

Error: HTTP 400 :: {"code":400,"message":"01071abb:3: Cannot create/modify published policy '/Common/Drafts/Axiom/my_policy' directly, try specifying a draft folder like '/Common/Drafts/Axiom/Drafts/my_policy'.","errorStack":[],"apiError":3}

  on main.tf line 7, in resource "bigip_ltm_policy" "test-policy":
   7: resource "bigip_ltm_policy" "test-policy" {
 
``` 

It seems like there's an issue with the creation of the draft policy. And this policy seems to be always created in /Common. That's why a user with manager rights wouldn't be able to do it. 

Tracking this internally with TER-476

**Comment by focrensh** [comment]
Quick update: This update/fix will not be in the upcoming release this week, but is estimated for the release afterwards. It should be available within a feature branch prior for testing. We will update this issue accordingly.

**Comment by papineni87** [comment]
@benjfranklin Partition issue is fixed and is available in master branch. Can you clone the latest master branch and check if it resolves your issue or not.

Terraform Config:


provider "bigip" {
 ...........................
}

resource "bigip_ltm_policy" "test-policy" {
  name           = "/test/13_policy"
  strategy       = "best-match"
  requires       = ["http"]
  controls       = ["forwarding"]
  rule {
      name = "Axiom_Environment_APP1_Redirect"

  condition {
        http_host = true
        contains = true
   //     value = true
        values = ["acme.example.com"]
        request = true
      }
   action {
        forward = true
        pool    = "/test/Axiom_Environment_APP12_Pool"
    }

   }

  depends_on = [bigip_ltm_pool.mypool]
}


resource "bigip_ltm_pool" "mypool" {
  name                = "/test/Axiom_Environment_APP12_Pool"
  allow_nat           = "yes"
  allow_snat          = "yes"
  load_balancing_mode = "round-robin"
}



terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # bigip_ltm_policy.test-policy will be created
  + resource "bigip_ltm_policy" "test-policy" {
      + controls = [
          + "forwarding",
        ]
      + id       = (known after apply)
      + name     = "/test/12_policy"
      + requires = [
          + "http",
        ]
      + strategy = "best-match"

      + rule {
          + name = "Axiom_Environment_APP1_Redirect"

          + action {
              + app_service          = (known after apply)
              + application          = (known after apply)
              + asm                  = (known after apply)
              + avr                  = (known after apply)
              + cache                = (known after apply)
              + carp                 = (known after apply)
              + category             = (known after apply)
              + classify             = (known after apply)
              + clone_pool           = (known after apply)
              + code                 = (known after apply)
              + compress             = (known after apply)
              + content              = (known after apply)
              + cookie_hash          = (known after apply)
              + cookie_insert        = (known after apply)
              + cookie_passive       = (known after apply)
              + cookie_rewrite       = (known after apply)
              + decompress           = (known after apply)
              + defer                = (known after apply)
              + destination_address  = (known after apply)
              + disable              = (known after apply)
              + domain               = (known after apply)
              + enable               = (known after apply)
              + expiry               = (known after apply)
              + expiry_secs          = (known after apply)
              + expression           = (known after apply)
              + extension            = (known after apply)
              + facility             = (known after apply)
              + forward              = true
              + from_profile         = (known after apply)
              + hash                 = (known after apply)
              + host                 = (known after apply)
              + http                 = (known after apply)
              + http_basic_auth      = (known after apply)
              + http_cookie          = (known after apply)
              + http_header          = (known after apply)
              + http_referer         = (known after apply)
              + http_reply           = (known after apply)
              + http_set_cookie      = (known after apply)
              + http_uri             = (known after apply)
              + ifile                = (known after apply)
              + insert               = (known after apply)
              + internal_virtual     = (known after apply)
              + ip_address           = (known after apply)
              + key                  = (known after apply)
              + l7dos                = (known after apply)
              + length               = (known after apply)
              + location             = (known after apply)
              + log                  = (known after apply)
              + ltm_policy           = (known after apply)
              + member               = (known after apply)
              + message              = (known after apply)
              + netmask              = (known after apply)
              + nexthop              = (known after apply)
              + node                 = (known after apply)
              + offset               = (known after apply)
              + path                 = (known after apply)
              + pem                  = (known after apply)
              + persist              = (known after apply)
              + pin                  = (known after apply)
              + policy               = (known after apply)
              + pool                 = "/test/Axiom_Environment_APP12_Pool"
              + port                 = (known after apply)
              + priority             = (known after apply)
              + profile              = (known after apply)
              + protocol             = (known after apply)
              + query_string         = (known after apply)
              + rateclass            = (known after apply)
              + redirect             = (known after apply)
              + remove               = (known after apply)
              + replace              = (known after apply)
              + request              = (known after apply)
              + request_adapt        = (known after apply)
              + reset                = (known after apply)
              + response             = (known after apply)
              + response_adapt       = (known after apply)
              + scheme               = (known after apply)
              + script               = (known after apply)
              + select               = (known after apply)
              + server_ssl           = (known after apply)
              + set_variable         = (known after apply)
              + snat                 = (known after apply)
              + snatpool             = (known after apply)
              + source_address       = (known after apply)
              + ssl_client_hello     = (known after apply)
              + ssl_server_handshake = (known after apply)
              + ssl_server_hello     = (known after apply)
              + ssl_session_id       = (known after apply)
              + status               = (known after apply)
              + tcl                  = (known after apply)
              + tcp_nagle            = (known after apply)
              + text                 = (known after apply)
              + timeout              = (known after apply)
              + tm_name              = (known after apply)
              + uie                  = (known after apply)
              + universal            = (known after apply)
              + value                = (known after apply)
              + virtual              = (known after apply)
              + vlan                 = (known after apply)
              + vlan_id              = (known after apply)
              + wam                  = (known after apply)
              + write                = (known after apply)
            }

          + condition {
              + address                 = (known after apply)
              + all                     = (known after apply)
              + app_service             = (known after apply)
              + browser_type            = (known after apply)
              + browser_version         = (known after apply)
              + case_insensitive        = (known after apply)
              + case_sensitive          = (known after apply)
              + cipher                  = (known after apply)
              + cipher_bits             = (known after apply)
              + client_ssl              = (known after apply)
              + code                    = (known after apply)
              + common_name             = (known after apply)
              + contains                = true
              + continent               = (known after apply)
              + country_code            = (known after apply)
              + country_name            = (known after apply)
              + cpu_usage               = (known after apply)
              + device_make             = (known after apply)
              + device_model            = (known after apply)
              + domain                  = (known after apply)
              + ends_with               = (known after apply)
              + equals                  = (known after apply)
              + expiry                  = (known after apply)
              + extension               = (known after apply)
              + external                = (known after apply)
              + geoip                   = (known after apply)
              + greater                 = (known after apply)
              + greater_or_equal        = (known after apply)
              + host                    = (known after apply)
              + http_basic_auth         = (known after apply)
              + http_cookie             = (known after apply)
              + http_header             = (known after apply)
              + http_host               = true
              + http_method             = (known after apply)
              + http_referer            = (known after apply)
              + http_set_cookie         = (known after apply)
              + http_status             = (known after apply)
              + http_uri                = (known after apply)
              + http_user_agent         = (known after apply)
              + http_version            = (known after apply)
              + index                   = (known after apply)
              + internal                = (known after apply)
              + isp                     = (known after apply)
              + last_15secs             = (known after apply)
              + last_1min               = (known after apply)
              + last_5mins              = (known after apply)
              + less                    = (known after apply)
              + less_or_equal           = (known after apply)
              + local                   = (known after apply)
              + major                   = (known after apply)
              + matches                 = (known after apply)
              + minor                   = (known after apply)
              + missing                 = (known after apply)
              + mss                     = (known after apply)
              + not                     = (known after apply)
              + org                     = (known after apply)
              + password                = (known after apply)
              + path                    = (known after apply)
              + path_segment            = (known after apply)
              + port                    = (known after apply)
              + present                 = (known after apply)
              + protocol                = (known after apply)
              + query_parameter         = (known after apply)
              + query_string            = (known after apply)
              + region_code             = (known after apply)
              + region_name             = (known after apply)
              + remote                  = (known after apply)
              + request                 = true
              + response                = (known after apply)
              + route_domain            = (known after apply)
              + rtt                     = (known after apply)
              + scheme                  = (known after apply)
              + server_name             = (known after apply)
              + ssl_cert                = (known after apply)
              + ssl_client_hello        = (known after apply)
              + ssl_extension           = (known after apply)
              + ssl_server_handshake    = (known after apply)
              + ssl_server_hello        = (known after apply)
              + starts_with             = (known after apply)
              + tcp                     = (known after apply)
              + text                    = (known after apply)
              + tm_name                 = (known after apply)
              + unnamed_query_parameter = (known after apply)
              + user_agent_token        = (known after apply)
              + username                = (known after apply)
              + value                   = (known after apply)
              + values                  = [
                  + "acme.example.com",
                ]
              + version                 = (known after apply)
              + vlan                    = (known after apply)
              + vlan_id                 = (known after apply)
            }
        }
    }

  # bigip_ltm_pool.mypool will be created
  + resource "bigip_ltm_pool" "mypool" {
      + allow_nat           = "yes"
      + allow_snat          = "yes"
      + id                  = (known after apply)
      + load_balancing_mode = "round-robin"
      + monitors            = (known after apply)
      + name                = "/test/Axiom_Environment_APP12_Pool"
      + reselect_tries      = (known after apply)
      + service_down_action = (known after apply)
      + slow_ramp_time      = (known after apply)
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

bigip_ltm_pool.mypool: Creating...
bigip_ltm_pool.mypool: Creation complete after 2s [id=/test/Axiom_Environment_APP12_Pool]
bigip_ltm_policy.test-policy: Creating...
bigip_ltm_policy.test-policy: Creation complete after 3s [id=/test/12_policy]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

**Comment by papineni87** [comment]
Issue is resolved in 1.4 release.
